packages/xo-server/src/xo-mixins/resource-sets.js

```diff
@@ -171,6 +171,20 @@ export default class {
     const store = this._store
 
     if (await store.has(id)) {
+      await Promise.all(
+        mapToArray(this._xo.getAllXapis(), xapi =>
```

Comment:
The resource set is a XO logic, so, we can manipulate XO objects instead of XAPI objects.

To do that, you can use this [XO method](https://github.com/vatesfr/xen-orchestra/blob/master/packages/xo-server/src/xo.js#L83) which allows to pass a filter.